### ISSUE TITLE
Add a show_versions function

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/geopandas#899]
Similarly to `pandas.show_versions()` and `sklearn.show_versions()`, it would be nice to have a function like that for GeoPandas. We can probably base the code on those examples. 

It could list the versions of the direct python dependencies, but also try to show the versions of the underlying GEOS / GDAL libraries.

**Comment by shakasom** [comment]
I am trying to implement this but I need guidance since this is my first contribution. Where would that function fit? I actually made the basic printing functions and it works in my local machine but I am also trying to figure out how to commit and push this.

**Comment by stevenbw** [comment]
@jorisvandenbossche is there anything I can help with here?

**Comment by jorisvandenbossche** [comment]
@ocefpaf @sgillies 

For shapely, we can find information about the GEOS it is build against with (or at least, the installed GEOS it finds at run time, I suppose):

```
In [1]: import shapely._buildcfg

In [2]: shapely._buildcfg.geos_version 
Out[2]: (3, 7, 1)

In [3]: shapely._buildcfg.geos_library_path                     
Out[3]: '/home/joris/miniconda3/envs/dev37/lib/libgeos_c.so'
```

Is that the appropriate API to use?

For fiona, I can do something like:

```
In [7]: import fiona.env                                    

In [8]: fiona.env.get_gdal_release_name()                                                   
Out[8]: '2.4.0'
```

**Comment by ocefpaf** [comment]
I usually do `shapely._buildcfg.geos_version` b/c I like the tuple response in case I need to check only MAJOR and or MINOR, for that reason I use `fiona.gdal_version` instead of `fiona.env.get_gdal_release_name()`.

**Comment by martinfleis** [comment]
Even though show_versions has been just merged in #918 I will keep this open as a reminder to add GEOS, GDAL an PROJ versions checking in the future.
